lib/matplotlib/ticker.py

```diff
@@ -1170,25 +1170,181 @@ class LogitFormatter(Formatter):
     """
     Probability formatter (using Math text).
     """
-    def __call__(self, x, pos=None):
-        s = ''
-        if 0.01 <= x <= 0.99:
-            s = '{:.2f}'.format(x)
-        elif x < 0.01:
-            if is_decade(x):
-                s = '$10^{{{:.0f}}}$'.format(np.log10(x))
-            else:
-                s = '${:.5f}$'.format(x)
-        else:  # x > 0.99
-            if is_decade(1-x):
-                s = '$1-10^{{{:.0f}}}$'.format(np.log10(1-x))
+
+    def __init__(
+        self,
+        *,
+        use_overline=False,
+        minor=False,
+        minor_threshold=25,
+        minor_number=4,
+    ):
+        r"""
+        Parameters
+        ----------
+        use_overline : bool (default: False)
+            if x > 1/2, with x = 1-v, indicate if x should be displayed as
+            $\overline{v}$. The default is to display $1-v$.
+
+        minor : bool (default: False)
+            indicate if the formatter is formatting minor ticks or not.
+            Basically minor ticks are not labelled, except when only few ticks
+            are provided, the most espaced ticks are labelled. See others
+            parameters to change the default behavior.
+
+        minor_threshold : int (default: 25)
+            maximum number of locs for labelling some minor ticks. This
+            parameter have no effect if minor is False.
+
+        minor_number : int (default: 4)
+            number of ticks which are labelled when the number of ticks is
+            below the threshold.
+        """
+        self._use_overline = use_overline
+        self._minor = minor
+        self._labelled = set()
+        self._minor_threshold = minor_threshold
+        self._minor_number = minor_number
+
+    def use_overline(self, use_overline):
+        r"""
+        Switch display mode with overline for labelling p>1/2
+
+        Parameters
+        ----------
+        use_overline : bool (default: False)
+            if x > 1/2, with x = 1-v, indicate if x should be displayed as
+            $\overline{v}$. The default is to display $1-v$.
+        """
+        self._use_overline = use_overline
+
+    def label_minor(self, minor):
```

Comment:
Could this be basically folded into `set_minor_threshold(0)`? (not sure it's a nice API though)

Reply:
Not exactly,

 - if `_minor` is `True`, indicates that this formatter is used for minor ticks. Therefore, some ticks (spaced ticks as LogFormatter sublabels) are labelled.
 - if `_minor` is `False`, indicates that this formatter is used for major ticks. Therefore, all ticks are labelled.

`set_minor_threshold(0)` implies when `_minor` is `True` that no minor ticks is labelled.

I understand the difficulty. I thing the `label_minor` can be deleted. The `_minor` attr should be set only at the init.

I propose to delete the method `label_minor`. What do you think?

Reply:
sounds reasonable.

Reply:
Ok, I set the default value in both.

Reply:
Done.